hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -163,6 +165,25 @@ else if( "myisam".equals( storageEngine.toLowerCase() ) ) {
 		getDefaultProperties().setProperty( Environment.STATEMENT_BATCH_SIZE, DEFAULT_BATCH_SIZE );
 
 		uniqueDelegate = new MySQLUniqueDelegate( this );
+		sizeStrategy = new SizeStrategyImpl() {
+			@Override
+			public Size resolveSize(
+					SqlTypeDescriptor sqlType,
+					JavaTypeDescriptor javaType,
+					Integer precision,
+					Integer scale,
+					Long length) {
+				final int jdbcTypeCode = sqlType.getSqlType();
+				switch ( jdbcTypeCode ) {
+					case Types.BIT:
+						// MySQL allows BIT with a length up to 64
+						if ( length != null ) {
+							return Size.length( Math.min( Math.max( length, 1 ), 64 ) );
+						}
```

Comment:
Do you think this is really necessary? It's not something we do with other types, nor in other dialects.

What's the reason for doing it specifically for this one case?

Reply:
Since MySQL uses the bit type for booleans, and the JPA default when using the `@Column` annotation is 255, this might be relevant, although I think I handled this already in the `Dialect`, so it might be unnecessary to keep this around, although I think if we want to support the bit type, it might make sense to still keep this around.